### ISSUE TITLE
[Sprint] Implement Optimized 1D Convolutional Neural Network (CNN)

### ISSUE THREAD

**Issue by ankit02327** [ankit02327/stock-price#117]
As part of the Algorithmic & ML Sprint, we need to implement a production-ready, optimized Convolutional Neural Network (CNN) for stock price prediction in `backend/algorithms/optimised/cnn/cnn.py`.

The current scratch implementation relies on `Conv2D` layers intended for image classification. For tabular financial time-series data, we must switch to a `Conv1D` architecture. This requires reshaping our standard 2D tabular feature matrix `(n_samples, n_features)` into a 3D time-windowed tensor `(n_samples, window_size, n_features)` so the convolutional filters can slide across the time axis.

**Tasks:**
- [ ] Create `cnn.py` inside `backend/algorithms/optimised/cnn/` and implement the `CNNModel` class inheriting from `ModelInterface`.
- [ ] Implement a sliding window data generator within `fit()` and `predict()` to reshape the 2D tabular `X` matrix into a 3D tensor suitable for `Conv1D`.
- [ ] Build a Keras `Sequential` model utilizing `Conv1D`, `MaxPooling1D`, `Flatten`, and `Dense` layers to extract temporal features.
- [ ] Explicitly enforce `shuffle=False` during Keras `fit()` to prevent time-series data leakage.

**Comment by ankit02327** [comment]
1. **Time-Window Reshaping:** Create a helper function inside `CNNModel` that iterates over `X` and creates overlapping windows of a specific `window_size` (e.g., 10 days). If `X` is `(1000, 15)`, the reshaped `X_3d` should be `(990, 10, 15)`.
2. **Conv1D Architecture:** Use `tensorflow.keras.layers.Conv1D` with filters=64 and kernel_size=3. Follow it with `MaxPooling1D`, `Flatten`, and a regression `Dense(1)` head.
3. **Data Integrity:** Just like the ANN and Autoencoder, ensure `shuffle=False` is passed to `self.model.fit()` to prevent future data from leaking into the training batches.

**Comment by ankit02327** [comment]
```py
import numpy as np
from tensorflow.keras.models import Sequential
from tensorflow.keras.layers import Conv1D, MaxPooling1D, Flatten, Dense, Dropout
from sklearn.preprocessing import RobustScaler

def create_time_windows(X, y=None, window_size=10):
    """Reshape 2D tabular data into 3D sequential data for Conv1D"""
    X_3d = []
    y_target = [] if y is not None else None
    
    for i in range(len(X) - window_size):
        X_3d.append(X[i:(i + window_size)])
        if y is not None:
            y_target.append(y[i + window_size])
            
    if y is not None:
        return np.array(X_3d), np.array(y_target)
    return np.array(X_3d)

# Inside CNNModel.fit():
self.scaler = RobustScaler()
X_scaled = self.scaler.fit_transform(X)

X_3d, y_target = create_time_windows(X_scaled, y, window_size=10)

self.model = Sequential([
    Conv1D(filters=64, kernel_size=3, activation='relu', input_shape=(10, X.shape[1])),
    MaxPooling1D(pool_size=2),
    Dropout(0.2),
    Conv1D(filters=32, kernel_size=2, activation='relu'),
    Flatten(),
    Dense(32, activation='relu'),
    Dense(1, activation='linear')
])

self.model.compile(optimizer='adam', loss='mse', metrics=['mae'])

self.model.fit(
    X_3d, y_target,
    epochs=100,
    batch_size=32,
    validation_split=0.2,
    shuffle=False,  # Critical for time-series integrity
    verbose=0
)
```

**Comment by AKASHPANDEY4642** [comment]
I want to try this.Can you assign me this issue

**Comment by ankit02327** [comment]
ok @AKASHPANDEY4642 I am assigning you this issue.
